auth-input.component.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, forwardRef, Input, OnInit} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+  selector: 'app-auth-input',
+  template:
+      `
+    <mat-form-field class="full-width" appearance="fill">
+      <mat-label>{{type==='email'?'Email':'Password'}}</mat-label>
+      <input [type]="type" matInput [placeholder]="type==='email'?'username@gmail.com':''" required="true" [(ngModel)]="value">
+    </mat-form-field>
+  `,
+  styles: [
+    `
+      .full-width{
+        width: 100%;
+      }
+    `

```

Comment:
use auth-input.component.html for template
use auth-input.component.scss for styling